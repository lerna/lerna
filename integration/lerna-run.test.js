"use strict";

const cliRunner = require("@lerna-test/cli-runner");
const initFixture = require("@lerna-test/init-fixture")(__dirname);

/**
 * NOTE: We do not test the "missing test script" case here
 * because Windows makes the snapshots impossible to stabilize.
 */
describe("lerna run", () => {
  test("my-script --scope", async () => {
    const cwd = await initFixture("lerna-run");
    const args = [
      "run",
      "my-script",
      "--scope=package-1",
      "--concurrency=1",
      // args below tell npm to be quiet
      "--",
      "--silent",
    ];
    const { stdout, stderr } = await cliRunner(cwd)(...args);
    expect(stdout).toBe("package-1");
    expect(stderr).toMatchSnapshot("stderr");
  });

  test("test --ignore", async () => {
    const cwd = await initFixture("lerna-run");
    const args = [
      "run",
      "--concurrency=1",
      "test",
      "--ignore",
      "package-@(1|2|3)",
      // args below tell npm to be quiet
      "--",
      "--silent",
    ];
    const { stdout, stderr } = await cliRunner(cwd)(...args);
    expect(stdout).toBe("package-4");
    expect(stderr).toMatchSnapshot("stderr");
  });

  test("test --stream", async () => {
    const cwd = await initFixture("lerna-run");
    const args = [
      "run",
      "--stream",
      "test",
      "--concurrency=1",
      // args below tell npm to be quiet
      "--",
      "--silent",
    ];
    const { stdout, stderr } = await cliRunner(cwd)(...args);
    expect(stdout).toMatchSnapshot("stdout");
    expect(stderr).toMatchSnapshot("stderr");
  });

  test("test --parallel", async () => {
    const cwd = await initFixture("lerna-run");
    const args = [
      "run",
      "test",
      "--parallel",
      // args below tell npm to be quiet
      "--",
      "--silent",
    ];
    const { stdout, stderr } = await cliRunner(cwd)(...args);
    expect(stderr).toMatchSnapshot("stderr");

    // order is non-deterministic, so assert each item seperately
    expect(stdout).toMatch("package-1: package-1");
    expect(stdout).toMatch("package-2: package-2");
    expect(stdout).toMatch("package-3: package-3");
    expect(stdout).toMatch("package-4: package-4");
  });

  test("my-script --parallel", async () => {
    const cwd = await initFixture("lerna-run");
    const args = [
      "run",
      "--parallel",
      "my-script",
      // args below tell npm to be quiet
      "--",
      "--silent",
    ];
    const { stdout, stderr } = await cliRunner(cwd)(...args);
    expect(stderr).toMatchSnapshot("stderr");

    // order is non-deterministic, so assert each item seperately
    expect(stdout).toMatch("package-1: package-1");
    expect(stdout).not.toMatch("package-2");
    expect(stdout).toMatch("package-3: package-3");
    expect(stdout).not.toMatch("package-4");
  });
});
