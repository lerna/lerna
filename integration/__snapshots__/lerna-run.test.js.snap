// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lerna run my-script --parallel: stderr 1`] = `
lerna info version __TEST_VERSION__
lerna info run in 2 package(s): npm run my-script --silent
lerna success run Ran npm script 'my-script' in packages:
lerna success - package-1
lerna success - package-3
`;

exports[`lerna run my-script --scope: stderr 1`] = `
lerna info version __TEST_VERSION__
lerna info scope package-1
lerna success run Ran npm script 'my-script' in packages:
lerna success - package-1
`;

exports[`lerna run test --ignore: stderr 1`] = `
lerna info version __TEST_VERSION__
lerna info ignore package-@(1|2|3)
lerna success run Ran npm script 'test' in packages:
lerna success - package-4
`;

exports[`lerna run test --parallel: stderr 1`] = `
lerna info version __TEST_VERSION__
lerna info run in 4 package(s): npm run test --silent
lerna success run Ran npm script 'test' in packages:
lerna success - package-1
lerna success - package-2
lerna success - package-3
lerna success - package-4
`;

exports[`lerna run test --stream: stderr 1`] = `
lerna info version __TEST_VERSION__
lerna success run Ran npm script 'test' in packages:
lerna success - package-1
lerna success - package-2
lerna success - package-3
lerna success - package-4
`;

exports[`lerna run test --stream: stdout 1`] = `
package-3: package-3
package-4: package-4
package-1: package-1
package-2: package-2
`;
